x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -499,6 +501,8 @@ public LogicalPlan apply(LogicalPlan plan) {
 
             plan = plan.transformUp(p -> {
                 // Apply the replacement inside Filter and Eval (which shouldn't make a difference)
+                // TODO: also allow aggregates once aggs on constants are supported.
+                // C.f. https://github.com/elastic/elasticsearch/issues/100634
```

Comment:
To be useful with aggs, we'd need to move `PropagateEvalFoldables` into the `Substitutions` batch and run this more than once - but we can worry about that later.